tests/chainermn_tests/datasets_tests/test_scatter.py

```diff
@@ -39,13 +40,25 @@ def check_scatter_dataset(self, original_dataset, shuffle=False, root=0):
             # Test the content of scattered datasets
             joined_dataset = sum((sub_dataset[:]
                                   for sub_dataset in sub_datasets), [])
+
+            # NOTE: values in original_dataset must be int-like type
```

Comment:
I am not sure if I understand, does it mean the values in `original_dataset` and `joined_dataset` must be casted to `int`, because `chainerx` and `numpy` have different behaviors when calling `type()`?

Reply:
Yes, their behaviors are different. Actually, ChainerX and CuPy have the same behavior, which is different from numpy.

```
>>> a = numpy.array([0])
>>> a
array([0])
>>> a[0]
0
>>> type(a[0])
<class 'numpy.int64'>         # <============ (A)

>>> b = chainerx.array([0])
>>> b
array([0], shape=(1,), dtype=int64, device='native:0')
>>> b[0]
array(0, shape=(), dtype=int64, device='native:0')
>>> type(b[0])
<class 'chainerx.ndarray'>   # <============ (B)
```

(A) and (B) show the difference. Because of this, we need to apply `int()` to get the same values between chainerx and numpy.

Reply:
I see, thank you. Could you rephrase the sentence "values in original_dataset must be int-like type". I think it is a little confusing.
My suggestion is "The values in original_dataset and joined_dataset must be casted to int"

Reply:
Added comment.